types/schema/data/Property.ts

```diff
@@ -96,20 +98,40 @@ type PropertyTypeMap = {
  */
 export type PropertyType = PropertyTypeMap[keyof PropertyTypeMap];
 
-/** @todo in future value & description should check against PlanningConstraints enum, but also allow custom per-council variables ?? */
-interface BasePlanningConstraint {
-  value: string;
-  description: string;
-}
+type PlanningConstraintKeys = keyof typeof PlanningConstraints;
 
-interface NonOverlappingPlanningConstraint extends BasePlanningConstraint {
+type GenericPlanningConstraint<TKey extends PlanningConstraintKeys> = {
+  value: TKey;
+  description: (typeof PlanningConstraints)[TKey];
+  category?:
+    | 'Ecology'
+    | 'Flooding'
+    | 'General policy'
+    | 'Heritage and conservation'
+    | 'Trees';
```

Comment:
I'm not convinced `category` is particularly meaningful to include, but BOPS is currently parsing last I heard so including this for migration feature parity